### DISCUSSION TITLE
Rails app running OOM, however operation works in CLI

### DISCUSSION THREAD

**Discussion by zyqxd** [minimagick/minimagick#573] [Q&A] [answered]
My goal is to resize the filesize of an image. I noticed PNGs take considerably more memory than JPEGs. I'd like to be able to resize these without causing my small rails app to fail (2gb memory). Whilst testing I noticed that while my console runs of out memory, if I do the same command in CLI it works file.

Here is my test inside rails console commands:

```ruby
pry(main)> image = MiniMagick::Image.open(url)
=> #<MiniMagick::Image:0x00007f9454c95910
 @info=
  #<MiniMagick::Image::Info:0x00007f9454c958c0
   @info={},
   @path="/app/tmp/mini_magick20240704-2733477-rw0c8.png">,
 @path="/app/tmp/mini_magick20240704-2733477-rw0c8.png",
 @tempfile=#<File:/app/tmp/mini_magick20240704-2733477-rw0c8.png (closed)>>
pry(main)> image.resize '120000@'
Killed
```

Here is the same command using the tmp file 
```
bash-4.4$ ls -al mini_magick20240704-2733477-rw0c8.png
-rw------- 1 app 64999 7006437 Jul  4 21:14 mini_magick20240704-2733477-rw0c8.png
bash-4.4$ /app/service/bin/mogrify -resize 120000@ mini_magick20240704-2733477-rw0c8.png
bash-4.4$ ls -al mini_magick20240704-2733477-rw0c8.png
-rw-r--r-- 1 app 64999 44149 Jul  4 21:40 mini_magick20240704-2733477-rw0c8.png
```

This is the actual file:
![original](https://github.com/minimagick/minimagick/assets/3641047/38c98867-8b20-456e-ae4d-a918d71b65f9)

I've also tried increasing some limits, to no avail. Example:
```ruby
image.combine_options do |opt|
  opt.limit 'area', '0'
  opt.limit 'disk', '8GiB'
  opt.limit 'map', '256MiB'
  opt.limit 'memory', '128MiB'
  opt.resize '120000@'
end
```
Yields:
```
mogrify: CacheResourcesExhausted `/app/tmp/mini_magick20240704-2733477-i19zqk.png' @ error/cache.c/OpenPixelCache/3860.
```

This may be a deeper issue than the gem, but any support for this is greatly appreciated

**Comment by janko** [comment]
The resize IM command likely doesn't use up more memory when it's run within the console, but perhaps combined with the memory usage the rails console itself takes hits the memory limits. For example, my rails console takes up 215mb of memory, which would be ~10% of your server's memory.

On my MacBook Air, I've observed the resize command taking ~800mb. So, combined with my rails console it would already be 50% of total memory. Not sure what else is running on your server, but maybe it's close to filling up the other 50%. I can just guess, though.

Moving this to a discussion, since it's unlikely an issue with MiniMagick.

**Reply by zyqxd** [reply]
Thanks, I think this is the case as well. Was really strange why I couldn't increase the resource limit but turns out you can only tighten outside of editing policy.xml

**Comment by zyqxd** [comment, marked as answer]
Found that imagemagick limit option can only tighten resource configuration, and cannot relax them:

> Note, you can restrict limits relative to any [security policies](https://imagemagick.org/source/policy-open.xml), but you cannot relax them.

https://imagemagick.org/script/command-line-options.php#limit
